App/src/main/java/cc/softwarefactory/lokki/android/activities/MainActivity.java

```diff
@@ -232,6 +236,29 @@ public void onNavigationDrawerItemSelected(int position) {
                 fragmentManager.beginTransaction().replace(R.id.container, new AboutFragment(), TAG_ABOUT_FRAGMENT).commit();
                 break;
 
+            case logoutPos: // Log Out
+                // Keep a reference to this so we can restart the activity from a callback
+                final MainActivity mainActivity = this;
+                // create the logout confirmation dialog
+                new AlertDialog.Builder(this)
+                        .setIcon(R.drawable.ic_power_settings_new_black_48dp)
+                        .setMessage(R.string.confirm_logout)
+                        .setPositiveButton(R.string.yes, new DialogInterface.OnClickListener() {
+                                    @Override
+                                    public void onClick(DialogInterface dialog, int which){
+                                        //Clear logged in status
+                                        PreferenceUtils.setString(mainActivity, PreferenceUtils.KEY_USER_ACCOUNT, null);
+                                        PreferenceUtils.setString(mainActivity, PreferenceUtils.KEY_USER_ID, null);
+                                        PreferenceUtils.setString(mainActivity, PreferenceUtils.KEY_AUTH_TOKEN, null);
```

Comment:
You would also need to clear the corresponding static fields in `MainApplication` in order to sign the user out completely.